<a href="https://colab.research.google.com/github/pancenaunov76/pancenaunov76/blob/main/homework_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Забелешка: ДОКОЛКУ НЕ Е НАПИШАНО ДА СЕ КОРИСТИ GROUPBY НЕ ТРЕБА ДА ГО КОРИСТИТЕ (ќе се смета за грешка)**

Import pandas

In [2]:
import pandas as pd
import numpy as np

1. Креирајте серија преку листа која има 6 елементи. Потоа променете ги индексите да бидат буквите: A, B, C, D, E, F.

In [2]:
l = [4, 5, 8, 9, 3, 2]
s = pd.Series(l)
s.index = ['A', 'B', 'C', 'D', 'E', 'F']
s

A    4
B    5
C    8
D    9
E    3
F    2
dtype: int64

2. Креирајте серија преку речник. Потоа променете ги индексите во вредности по ваш избор. Испринтајте ја серијата 2 пати, еднаш по самото креирање и еднаш по промената на индексите.

In [3]:
dicti = {'A':[1, 7, 6]}
s2 = pd.Series(dicti)
print(s2)
s2.index = ['B']
s2

A    [1, 7, 6]
dtype: object


B    [1, 7, 6]
dtype: object

3. Креирајте празен dataframe

In [4]:
df = pd.DataFrame()
df

Empty DataFrame
Columns: []
Index: []

4.1. Креирајте dataframe кој ќе има 3 колони: Име, Плата, Матичен Број. Пополнете ја табелата да има 10 редици. (име-стринг, плата-стринг (со $ пред секоја вредност), матичен број-стринг)

In [5]:
d = {'Ime':['Petar', 'Gorgi', 'Dragan', 'Zoran', 'Igor', 'Irena', 'Mia', 'Marija', 'Gurga', 'Jana'], 'Plata':['$10', '$20', '$30', '$40', '$50', '$60', '$70', '$800', '$9000', '$1000'], 'Maticen_Broj':['416516166141', '5115165116', '987882417468', '5150001141', '65416545', '654651516', '89797879', '321311313', '963936993', '4941191991']}
d
df = pd.DataFrame(d)
df

Ime  Plata  Maticen_Broj
0   Petar    $10  416516166141
1   Gorgi    $20    5115165116
2  Dragan    $30  987882417468
3   Zoran    $40    5150001141
4    Igor    $50      65416545
5   Irena    $60     654651516
6     Mia    $70      89797879
7  Marija   $800     321311313
8   Gurga  $9000     963936993
9    Jana  $1000    4941191991

4.2 Додадете нов ред кој ќе ги има следните вредности: име-Иле, плата-$2000, матичен број-1234567890123

In [6]:
nov_red = {'Ime':'Ile', 'Plata':'$2000', 'Maticen_Broj':'1234567890123'}
df = df.append(nov_red,ignore_index=True)
df

<ipython-input-6-8d4f29a6da4e>:2: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  df = df.append(nov_red,ignore_index=True)


Ime  Plata   Maticen_Broj
0    Petar    $10   416516166141
1    Gorgi    $20     5115165116
2   Dragan    $30   987882417468
3    Zoran    $40     5150001141
4     Igor    $50       65416545
5    Irena    $60      654651516
6      Mia    $70       89797879
7   Marija   $800      321311313
8    Gurga  $9000      963936993
9     Jana  $1000     4941191991
10     Ile  $2000  1234567890123

4.3 Извадете ги сите информации за Иле.

In [7]:
df[df['Ime']=='Ile']

Ime  Plata   Maticen_Broj
10  Ile  $2000  1234567890123

4.4 Вредностите во колоната плата сменете ги да бидат цели броеви.

In [8]:
df['Plata'] = df['Plata'].str.replace('$','').astype(int)
df

<ipython-input-8-bc1ddf292e98>:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df['Plata'] = df['Plata'].str.replace('$','').astype(int)


Ime  Plata   Maticen_Broj
0    Petar     10   416516166141
1    Gorgi     20     5115165116
2   Dragan     30   987882417468
3    Zoran     40     5150001141
4     Igor     50       65416545
5    Irena     60      654651516
6      Mia     70       89797879
7   Marija    800      321311313
8    Gurga   9000      963936993
9     Jana   1000     4941191991
10     Ile   2000  1234567890123

4.5 Додадете колона во која ќе го напишете полот на секоја личност. Додадете колона во која ќе ги напишете годините на секоја личност.

In [9]:
df['Pol'] = ['male','male','male','male','male', 'female', 'female', 'female', 'female', 'female', 'male']
df['Vozrast']  = [30, 56, 25, 47, 60, 19, 29, 32, 31, 35, 36]
df

Ime  Plata   Maticen_Broj     Pol  Vozrast
0    Petar     10   416516166141    male       30
1    Gorgi     20     5115165116    male       56
2   Dragan     30   987882417468    male       25
3    Zoran     40     5150001141    male       47
4     Igor     50       65416545    male       60
5    Irena     60      654651516  female       19
6      Mia     70       89797879  female       29
7   Marija    800      321311313  female       32
8    Gurga   9000      963936993  female       31
9     Jana   1000     4941191991  female       35
10     Ile   2000  1234567890123    male       36

4.6 Извадете ги сите машки од датсетот.

In [10]:
df[df['Pol']=='male']

Ime  Plata   Maticen_Broj   Pol  Vozrast
0    Petar     10   416516166141  male       30
1    Gorgi     20     5115165116  male       56
2   Dragan     30   987882417468  male       25
3    Zoran     40     5150001141  male       47
4     Igor     50       65416545  male       60
10     Ile   2000  1234567890123  male       36

4.7 Извадете ги сите женски и сите кои имаат плата повисока од 500.

In [11]:
df[(df['Pol']=='female') & (df['Plata']>500)]

Ime  Plata Maticen_Broj     Pol  Vozrast
7  Marija    800    321311313  female       32
8   Gurga   9000    963936993  female       31
9    Jana   1000   4941191991  female       35

4.8 Извадете ги сите кои имаат плата различна од 2000.

In [12]:
df[~(df['Plata']==2000)]

Ime  Plata  Maticen_Broj     Pol  Vozrast
0   Petar     10  416516166141    male       30
1   Gorgi     20    5115165116    male       56
2  Dragan     30  987882417468    male       25
3   Zoran     40    5150001141    male       47
4    Igor     50      65416545    male       60
5   Irena     60     654651516  female       19
6     Mia     70      89797879  female       29
7  Marija    800     321311313  female       32
8   Gurga   9000     963936993  female       31
9    Jana   1000    4941191991  female       35

4.9 Извадете ги матичните броеви на сите кои имаат помалку од 30 години или плата пониска од 700.

In [13]:
df[(df['Vozrast']<30) | (df['Plata']<700)]

Ime  Plata  Maticen_Broj     Pol  Vozrast
0   Petar     10  416516166141    male       30
1   Gorgi     20    5115165116    male       56
2  Dragan     30  987882417468    male       25
3   Zoran     40    5150001141    male       47
4    Igor     50      65416545    male       60
5   Irena     60     654651516  female       19
6     Mia     70      89797879  female       29

4.10 Извадете ги имињата на сите женски.

In [14]:
df[df['Pol']=='female'].Ime


5     Irena
6       Mia
7    Marija
8     Gurga
9      Jana
Name: Ime, dtype: object

5. Симнете го датасетот и прочитајте го: [линк](https://drive.google.com/file/d/1CbghFo-6BZ5dtzARo7fCmF89k_WCV0XR/view?usp=sharing)

In [11]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [3]:
ioan = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/loan (1).csv")
ioan

Loan_ID Gender Married Dependents     Education Self_Employed  \
0    LP001015   Male     Yes          0      Graduate            No   
1    LP001022   Male     Yes          1      Graduate            No   
2    LP001031   Male     Yes          2      Graduate            No   
3    LP001035   Male     Yes          2      Graduate            No   
4    LP001051   Male      No          0  Not Graduate            No   
..        ...    ...     ...        ...           ...           ...   
362  LP002971   Male     Yes         3+  Not Graduate           Yes   
363  LP002975   Male     Yes          0      Graduate            No   
364  LP002980   Male      No          0      Graduate            No   
365  LP002986   Male     Yes          0      Graduate            No   
366  LP002989   Male      No          0      Graduate           Yes   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5720                  0       110.0             360.0   
1               3076               1500       126.0             360.0   
2               5000               1800       208.0             360.0   
3               2340               2546       100.0             360.0   
4               3276                  0        78.0             360.0   
..               ...                ...         ...               ...   
362             4009               1777       113.0             360.0   
363             4158                709       115.0             360.0   
364             3250               1993       126.0             360.0   
365             5000               2393       158.0             360.0   
366             9200                  0        98.0             180.0   

     Credit_History Property_Area  
0               1.0         Urban  
1               1.0         Urban  
2               1.0         Urban  
3               NaN         Urban  
4               1.0         Urban  
..              ...           ...  
362             1.0         Urban  
363             1.0         Urban  
364             NaN     Semiurban  
365             1.0         Rural  
366             1.0         Rural  

[367 rows x 12 columns]

5.1 Колку колони и колку редови има датасетот?

In [17]:
ioan.shape

(367, 12)

5.2 Извадете информации за датасетот (колку се празни вредности, каков тип на податок има во секоја колона и слично)

In [18]:
ioan.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 367 entries, 0 to 366
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            367 non-null    object 
 1   Gender             356 non-null    object 
 2   Married            367 non-null    object 
 3   Dependents         357 non-null    object 
 4   Education          367 non-null    object 
 5   Self_Employed      344 non-null    object 
 6   ApplicantIncome    367 non-null    int64  
 7   CoapplicantIncome  367 non-null    int64  
 8   LoanAmount         362 non-null    float64
 9   Loan_Amount_Term   361 non-null    float64
 10  Credit_History     338 non-null    float64
 11  Property_Area      367 non-null    object 
dtypes: float64(3), int64(2), object(7)
memory usage: 34.5+ KB


5.3 Извадете ги сите омажени/оженети личности од датасетот кои имаат завршено факултет.

In [19]:
ioan_ = ioan[(ioan['Married']=='Yes') & (ioan['Education']=='Graduate')]
ioan_

Loan_ID  Gender Married Dependents Education Self_Employed  \
0    LP001015    Male     Yes          0  Graduate            No   
1    LP001022    Male     Yes          1  Graduate            No   
2    LP001031    Male     Yes          2  Graduate            No   
3    LP001035    Male     Yes          2  Graduate            No   
8    LP001059    Male     Yes          2  Graduate           NaN   
..        ...     ...     ...        ...       ...           ...   
356  LP002935    Male     Yes          1  Graduate            No   
360  LP002965  Female     Yes          0  Graduate            No   
361  LP002969    Male     Yes          1  Graduate            No   
363  LP002975    Male     Yes          0  Graduate            No   
365  LP002986    Male     Yes          0  Graduate            No   

     ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0               5720                  0       110.0             360.0   
1               3076               1500       126.0             360.0   
2               5000               1800       208.0             360.0   
3               2340               2546       100.0             360.0   
8              13633                  0       280.0             240.0   
..               ...                ...         ...               ...   
356             3791               1936        85.0             360.0   
360             8550               4255        96.0             360.0   
361             2269               2167        99.0             360.0   
363             4158                709       115.0             360.0   
365             5000               2393       158.0             360.0   

     Credit_History Property_Area  
0               1.0         Urban  
1               1.0         Urban  
2               1.0         Urban  
3               NaN         Urban  
8               1.0         Urban  
..              ...           ...  
356             1.0         Urban  
360             NaN         Urban  
361             1.0     Semiurban  
363             1.0         Urban  
365             1.0         Rural  

[176 rows x 12 columns]

5.4 Која е личноста со највисок кредит?

In [20]:
a = ioan[ioan['LoanAmount']==ioan['LoanAmount'].max()]['Loan_ID'].values
print(f'Najvisok kredit ima liceto koe go zelo kreditot so id {a}.')

Najvisok kredit ima liceto koe go zelo kreditot so id ['LP001791'].


5.5 За секоја должина на исплаќање на кредит, пресметајте колку е највисокиот кредит?

In [99]:
period = ioan['Loan_Amount_Term'].unique()
print(period)
for i in period:
  #if i != 'NaN':
  kredit = ioan[ioan['Loan_Amount_Term']==i]['LoanAmount'].max()
  print(f'За должина на исплаќање на кредит од {i}, највисокиот кредит е {kredit}.')

[360. 240. 180.  nan  60. 480.  84.  12. 300. 350.  36. 120.   6.]
За должина на исплаќање на кредит од 360.0, највисокиот кредит е 550.0.
За должина на исплаќање на кредит од 240.0, највисокиот кредит е 280.0.
За должина на исплаќање на кредит од 180.0, највисокиот кредит е 185.0.
За должина на исплаќање на кредит од nan, највисокиот кредит е nan.
За должина на исплаќање на кредит од 60.0, највисокиот кредит е 139.0.
За должина на исплаќање на кредит од 480.0, највисокиот кредит е 149.0.
За должина на исплаќање на кредит од 84.0, највисокиот кредит е 132.0.
За должина на исплаќање на кредит од 12.0, највисокиот кредит е 260.0.
За должина на исплаќање на кредит од 300.0, највисокиот кредит е 152.0.
За должина на исплаќање на кредит од 350.0, највисокиот кредит е 133.0.
За должина на исплаќање на кредит од 36.0, највисокиот кредит е 118.0.
За должина на исплаќање на кредит од 120.0, највисокиот кредит е 80.0.
За должина на исплаќање на кредит од 6.0, највисокиот кредит е 95.0.


5.6 Од сите self employed личности во датасетот, извадете ги оние кои што имаат завршено факултет и имаат повеќе од 1 дете.

In [107]:
if ioan['Dependents'] != 'NaN':
  a = ioan['Dependents'].str.strip('+').astype(int)
  a

  b = ioan[(ioan['Education']=='Graduate') & (ioan['Dependents'] > 1)]['Self_Employed']
  b

ValueError: The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().

5.7 За секој тип на property пресметајте колку од луѓето кои земале кредите имаат завршено факултет и колку немаат.

In [63]:
property = ioan['Property_Area'].unique().tolist()
for i in property:
  so_fakultet = ioan[(ioan['Property_Area']==i) & (ioan['Education']=='Graduate')]['Loan_ID'].nunique()
  bez_fakultet = ioan[(ioan['Property_Area']==i)][ioan['Education']=='Not Graduate']['Loan_ID'].nunique()
  print(f'Za Property_Area-{i} ima {so_fakultet} luge so zavrsen fakultet i {bez_fakultet} luge bez zavrsen fakultet.\n')


Za Property_Area-Urban ima 111 luge so zavrsen fakultet i 29 luge bez zavrsen fakultet.

Za Property_Area-Semiurban ima 87 luge so zavrsen fakultet i 29 luge bez zavrsen fakultet.

Za Property_Area-Rural ima 85 luge so zavrsen fakultet i 26 luge bez zavrsen fakultet.



<ipython-input-63-5174db7f8c87>:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  bez_fakultet = ioan[(ioan['Property_Area']==i)][ioan['Education']=='Not Graduate']['Loan_ID'].nunique()


6. Симнете го датасетот и прочитајте го. [линк](https://drive.google.com/file/d/1ngQzMRq5vowcaNQLhIDEwJftXLbQorqj/view?usp=sharing)


In [4]:
city_df = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/city_day.csv")
city_df

City        Date  PM2.5   PM10     NO    NO2    NOx    NH3  \
0          Ahmedabad  2015-01-01    NaN    NaN   0.92  18.22  17.15    NaN   
1          Ahmedabad  2015-01-02    NaN    NaN   0.97  15.69  16.46    NaN   
2          Ahmedabad  2015-01-03    NaN    NaN  17.40  19.30  29.70    NaN   
3          Ahmedabad  2015-01-04    NaN    NaN   1.70  18.48  17.97    NaN   
4          Ahmedabad  2015-01-05    NaN    NaN  22.10  21.42  37.76    NaN   
...              ...         ...    ...    ...    ...    ...    ...    ...   
29526  Visakhapatnam  2020-06-27  15.02  50.94   7.68  25.06  19.54  12.47   
29527  Visakhapatnam  2020-06-28  24.38  74.09   3.42  26.06  16.53  11.99   
29528  Visakhapatnam  2020-06-29  22.91  65.73   3.45  29.53  18.33  10.71   
29529  Visakhapatnam  2020-06-30  16.64  49.97   4.05  29.26  18.80  10.03   
29530  Visakhapatnam  2020-07-01  15.00  66.00   0.40  26.85  14.05   5.20   

          CO    SO2      O3  Benzene  Toluene  Xylene   AQI    AQI_Bucket  
0       0.92  27.64  133.36     0.00     0.02    0.00   NaN           NaN  
1       0.97  24.55   34.06     3.68     5.50    3.77   NaN           NaN  
2      17.40  29.07   30.70     6.80    16.40    2.25   NaN           NaN  
3       1.70  18.59   36.08     4.43    10.14    1.00   NaN           NaN  
4      22.10  39.33   39.31     7.01    18.89    2.78   NaN           NaN  
...      ...    ...     ...      ...      ...     ...   ...           ...  
29526   0.47   8.55   23.30     2.24    12.07    0.73  41.0          Good  
29527   0.52  12.72   30.14     0.74     2.21    0.38  70.0  Satisfactory  
29528   0.48   8.42   30.96     0.01     0.01    0.00  68.0  Satisfactory  
29529   0.52   9.84   28.30     0.00     0.00    0.00  54.0  Satisfactory  
29530   0.59   2.10   17.05      NaN      NaN     NaN  50.0          Good  

[29531 rows x 16 columns]

In [36]:
city_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29531 entries, 0 to 29530
Data columns (total 16 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   City        29531 non-null  object 
 1   Date        29531 non-null  object 
 2   PM2.5       24933 non-null  float64
 3   PM10        18391 non-null  float64
 4   NO          25949 non-null  float64
 5   NO2         25946 non-null  float64
 6   NOx         25346 non-null  float64
 7   NH3         19203 non-null  float64
 8   CO          27472 non-null  float64
 9   SO2         25677 non-null  float64
 10  O3          25509 non-null  float64
 11  Benzene     23908 non-null  float64
 12  Toluene     21490 non-null  float64
 13  Xylene      11422 non-null  float64
 14  AQI         24850 non-null  float64
 15  AQI_Bucket  24850 non-null  object 
dtypes: float64(13), object(3)
memory usage: 3.6+ MB


6.1 Додадете нова колона која ќе се вика year и ќе ја има годината за која станува збор за секој ред.

In [115]:
city_df['Date'] = pd.to_datetime(city_df['Date'])
city_df['year'] = city_df['Date'].dt.year.astype(int)

city_df

City       Date  PM2.5   PM10     NO    NO2    NOx    NH3  \
0          Ahmedabad 2015-01-01    NaN    NaN   0.92  18.22  17.15    NaN   
1          Ahmedabad 2015-01-02    NaN    NaN   0.97  15.69  16.46    NaN   
2          Ahmedabad 2015-01-03    NaN    NaN  17.40  19.30  29.70    NaN   
3          Ahmedabad 2015-01-04    NaN    NaN   1.70  18.48  17.97    NaN   
4          Ahmedabad 2015-01-05    NaN    NaN  22.10  21.42  37.76    NaN   
...              ...        ...    ...    ...    ...    ...    ...    ...   
29526  Visakhapatnam 2020-06-27  15.02  50.94   7.68  25.06  19.54  12.47   
29527  Visakhapatnam 2020-06-28  24.38  74.09   3.42  26.06  16.53  11.99   
29528  Visakhapatnam 2020-06-29  22.91  65.73   3.45  29.53  18.33  10.71   
29529  Visakhapatnam 2020-06-30  16.64  49.97   4.05  29.26  18.80  10.03   
29530  Visakhapatnam 2020-07-01  15.00  66.00   0.40  26.85  14.05   5.20   

          CO    SO2      O3  Benzene  Toluene  Xylene   AQI    AQI_Bucket  \
0       0.92  27.64  133.36     0.00     0.02    0.00   NaN           NaN   
1       0.97  24.55   34.06     3.68     5.50    3.77   NaN           NaN   
2      17.40  29.07   30.70     6.80    16.40    2.25   NaN           NaN   
3       1.70  18.59   36.08     4.43    10.14    1.00   NaN           NaN   
4      22.10  39.33   39.31     7.01    18.89    2.78   NaN           NaN   
...      ...    ...     ...      ...      ...     ...   ...           ...   
29526   0.47   8.55   23.30     2.24    12.07    0.73  41.0          Good   
29527   0.52  12.72   30.14     0.74     2.21    0.38  70.0  Satisfactory   
29528   0.48   8.42   30.96     0.01     0.01    0.00  68.0  Satisfactory   
29529   0.52   9.84   28.30     0.00     0.00    0.00  54.0  Satisfactory   
29530   0.59   2.10   17.05      NaN      NaN     NaN  50.0          Good   

       year  
0      2015  
1      2015  
2      2015  
3      2015  
4      2015  
...     ...  
29526  2020  
29527  2020  
29528  2020  
29529  2020  
29530  2020  

[29531 rows x 17 columns]

6.2 За секоја година извадете ги имињата на градовите кои имале NO повисоко од 5.

6.3 Извадете ги имињата на градовите кои во 2018 години за CO имале вредност 0.89

6.4 За секој град пресметајте во кој месец (користете го полето Date) имал највисок NO2.

6.5 За кои градови имаме информации во датасетот?

6.6 Извадете ги сите елементи за кои NH3 не е празна вредности и S02 е поголемо од 20.

7. Симнете и прочитајте го датасетот: [линк](https://drive.google.com/file/d/1HbZN-k_FF2y3nm4dt3QTceA00J64VMco/view?usp=sharing)

In [6]:
df_stations = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/stations.csv")
df_stations

StationId                                     StationName  \
0       AP001                  Secretariat, Amaravati - APPCB   
1       AP002  Anand Kala Kshetram, Rajamahendravaram - APPCB   
2       AP003                      Tirumala, Tirupati - APPCB   
3       AP004                 PWD Grounds, Vijayawada - APPCB   
4       AP005          GVM Corporation, Visakhapatnam - APPCB   
..        ...                                             ...   
225     WB010                       Jadavpur, Kolkata - WBPCB   
226     WB011    Rabindra Bharati University, Kolkata - WBPCB   
227     WB012               Rabindra Sarobar, Kolkata - WBPCB   
228     WB013                       Victoria, Kolkata - WBPCB   
229     WB014              Ward-32 Bapupara, Siliguri - WBPCB   

                  City           State  Status  
0            Amaravati  Andhra Pradesh  Active  
1    Rajamahendravaram  Andhra Pradesh     NaN  
2             Tirupati  Andhra Pradesh     NaN  
3           Vijayawada  Andhra Pradesh     NaN  
4        Visakhapatnam  Andhra Pradesh  Active  
..                 ...             ...     ...  
225            Kolkata     West Bengal  Active  
226            Kolkata     West Bengal  Active  
227            Kolkata     West Bengal  Active  
228            Kolkata     West Bengal  Active  
229           Siliguri     West Bengal     NaN  

[230 rows x 5 columns]

7.1 За секој state пресметајте колку станици има.

In [7]:
df_stations['State'].value_counts()

Delhi             38
Haryana           29
Uttar Pradesh     26
Maharashtra       22
Karnataka         20
Madhya Pradesh    16
West Bengal       14
Rajasthan         10
Bihar             10
Kerala             8
Punjab             8
Gujarat            6
Telangana          6
Tamil Nadu         5
Andhra Pradesh     5
Odisha             2
Jharkhand          1
Assam              1
Chandigarh         1
Meghalaya          1
Mizoram            1
Name: State, dtype: int64

7.2 За секој град пресметајте колку станици има што се активни.

In [ ]:
df_stations[df_stations['Status']=='Active']['City'].value_counts()

8. Симнете го и прочитајте го датасетот: [линк](https://drive.google.com/file/d/1AddtE1MN6gW9dK6Q7eAzNgGjRU5DzSxw/view?usp=sharing)

In [13]:
df_skin_care= pd.read_csv("/content/drive/MyDrive/Colab Notebooks/skin_care.csv")
df_skin_care

Unnamed: 0  price                                              title  \
0              0   6000                       BeautyBio Rose Quartz Roller   
1              1   2800                             Kopari Starry Eye Balm   
2              2   7200     StriVectin SuperC Brighten  and  Correct Serum   
3              3   5900  Skin Gym Rose Quartz Facial Workout Set Nordst...   
4              4   9150  Slip Marble  and  Charcoal Pillowcase  and  Sl...   
...          ...    ...                                                ...   
3239        3239  12539  Origins GinZing UltraHydrating EnergyBoosting ...   
3240        3240  18202             dermalogica PreCleanse Balm  and  Mitt   
3241        3241  22247                           Skin Inc Pure Serum Mist   
3242        3242  27101  Clinique Repairwear Uplifting Firming Cream fo...   
3243        3243  27101  Clinique Repairwear Uplifting Firming Cream fo...   

      stars  vote  
0       4.7    75  
1       4.0    11  
2       4.6   271  
3       5.0     1  
4       5.0     2  
...     ...   ...  
3239    4.6     8  
3240    4.6   552  
3241    4.9    11  
3242    4.2   154  
3243    4.2   276  

[3244 rows x 5 columns]

8.1 Избришете ја колоната Unnamed:0 .

In [22]:
df_skin_care.drop('Unnamed: 0',axis=1,inplace=True)
df_skin_care

price                                              title  stars  vote
0      6000                       BeautyBio Rose Quartz Roller    4.7    75
1      2800                             Kopari Starry Eye Balm    4.0    11
2      7200     StriVectin SuperC Brighten  and  Correct Serum    4.6   271
3      5900  Skin Gym Rose Quartz Facial Workout Set Nordst...    5.0     1
4      9150  Slip Marble  and  Charcoal Pillowcase  and  Sl...    5.0     2
...     ...                                                ...    ...   ...
3239  12539  Origins GinZing UltraHydrating EnergyBoosting ...    4.6     8
3240  18202             dermalogica PreCleanse Balm  and  Mitt    4.6   552
3241  22247                           Skin Inc Pure Serum Mist    4.9    11
3242  27101  Clinique Repairwear Uplifting Firming Cream fo...    4.2   154
3243  27101  Clinique Repairwear Uplifting Firming Cream fo...    4.2   276

[3244 rows x 4 columns]

8.2 Со groupby за секоја цена на производите извадете која е средната вредност на ѕвезди кои ги има добиено.

In [24]:
df_skin_care.groupby('price')['stars'].mean()

price
600        5.00
1700       4.60
2400       3.00
2500       1.35
2600       4.25
           ... 
606711     3.00
626969     0.00
849261     0.00
1051466    0.00
1546868    0.00
Name: stars, Length: 679, dtype: float64

8.3 Барањето под 8.2 пресметајте го без groupby

8.4 Кој е продуктот со најголем број на гласови?

In [87]:
a = df_skin_care[df_skin_care['vote']==df_skin_care['vote'].max()]['title'].values[0]
a

'Estée Lauder Advanced Night Repair Synchronized MultiRecovery Complex Serum Duo USD 210 Value'

8.5 За секој можност на добиени ѕвезди пресметајте кој е најскапиот производ.

In [78]:
unique_stars = df_skin_care['stars'].unique().tolist()
#unique_stars
for i in unique_stars:
  max_price = df_skin_care[df_skin_care['stars']== i]['price'].max()
  #print(max_price)
  ime = df_skin_care[df_skin_care['price']== max_price]['title'].values[0]
  #print(ime)
  print(f"Za broj na dobieni dzvezdi {i}, najskapiot proizvod e '{ime}'")

Za broj na dobieni dzvezdi 4.7, najskapiot proizvod e 'Crème de la Mer Moisturizing Cream Grande 1497 Value'
Za broj na dobieni dzvezdi 4.0, najskapiot proizvod e 'La Prairie White Caviar Eye Extraordinaire'
Za broj na dobieni dzvezdi 4.6, najskapiot proizvod e 'Crème de la Mer Moisturizing Cream Grande 1497 Value'
Za broj na dobieni dzvezdi 5.0, najskapiot proizvod e 'La Prairie Platinum Rare Cell Night Elixir Rejuvenating Serum'
Za broj na dobieni dzvezdi 4.4, najskapiot proizvod e 'Sisley Supremÿa at Night Supreme AntiAging Skin Care Cream'
Za broj na dobieni dzvezdi 3.9, najskapiot proizvod e 'ZIIP Beauty Electrical Facial Device'
Za broj na dobieni dzvezdi 0.0, najskapiot proizvod e 'La Prairie Platinum Rare Haute Rejuvenation Set'
Za broj na dobieni dzvezdi 4.8, najskapiot proizvod e 'SUBLIMAGE LEXTRAIT 
'
Za broj na dobieni dzvezdi 4.2, najskapiot proizvod e 'La Prairie Skin Caviar Absolute Filler'
Za broj na dobieni dzvezdi 4.1, najskapiot proizvod e 'La Prairie White Caviar Ey

9. Симнете го датасетот и вчитајте го: [линк](https://drive.google.com/file/d/1bsXNgnpktSOgx6jwMLWt8Gxm2qAZ1ynz/view?usp=sharing)

In [41]:
df_cardio = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/cardio.csv")
df_cardio

id  age  education sex is_smoking  cigsPerDay  BPMeds  prevalentStroke  \
0    3390   43        2.0   M        YES        35.0     0.0                0   
1    3391   56        3.0   F         NO         0.0     0.0                0   
2    3392   58        1.0   F        YES        20.0     0.0                0   
3    3393   47        3.0   F         NO         0.0     0.0                0   
4    3394   44        1.0   M         NO         0.0     0.0                0   
..    ...  ...        ...  ..        ...         ...     ...              ...   
843  4233   46        3.0   F         NO         0.0     0.0                0   
844  4234   58        2.0   F         NO         0.0     0.0                0   
845  4235   55        1.0   M         NO         0.0     0.0                0   
846  4236   45        3.0   F         NO         0.0     0.0                0   
847  4237   56        NaN   M        YES        25.0     0.0                0   

     prevalentHyp  diabetes  totChol  sysBP  diaBP    BMI  heartRate  glucose  
0               0         0    207.0  117.0   65.0  24.42       60.0    100.0  
1               0         0    192.0  122.0   82.5  28.61       68.0     58.0  
2               1         0    260.0  180.0  100.0  25.56      100.0      NaN  
3               0         0    231.0  102.5   66.0  23.40       70.0     78.0  
4               0         0    160.0  118.5   87.0  25.81       54.0      NaN  
..            ...       ...      ...    ...    ...    ...        ...      ...  
843             0         0    277.0  122.5   77.5  27.42       63.0     77.0  
844             0         0    190.0  132.0   67.0  23.08       65.0     70.0  
845             0         0    199.0  134.0   80.0  26.41      120.0      NaN  
846             1         0    231.0  157.5  104.5  22.86       75.0     92.0  
847             0         0    255.0  138.0   80.0  23.44       67.0     79.0  

[848 rows x 16 columns]

9.1 Извадете ги сите кои имаат диабетис.

9.2 Вредностите на колоната is_smoking променете ги: сите што се YES да станат True, сите што се No да станат False.

9.3 За сите категории на едукација извадете го id-то на личноста што има највисок heart rate.

9.4 Со groupby пресметајте за секоја возраст колку е нависоката вредност за totChol.

9.5 Барањето под 9.4 пресметајте го без groupby.

9.6 За сите можности на пол пресметајте колку луѓе се пушачи и колку не се.

9.7 Извадете ги сите ids кои имаат diaBP поголемо од 70 и MI помало од 25.

9.8 Колку луѓе пушат по 20 цигари на ден?